model/pipeline/src/main/proto/beam_pipeline_metrics.proto

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing the Runner API, which is the runner-independent,
+ * SDK-independent definition of the Beam model.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.model.pipeline.v1;
+
+option go_package = "pipeline_v1";
+option java_package = "org.apache.beam.model.pipeline.v1";
+option java_outer_classname = "PipelineMetrics";
+
+message MetricKey {
+    string step = 1;
+    MetricName metric_name = 2;
+}
+
+message MetricName {
+    string namespace = 1;
+    string name = 2;
+}
+
+message Metric {
+    // (Required) The data for this metric.
+    oneof data {
+        CounterData counter_data = 1;
+        DistributionData distribution_data = 2;
+        ExtremaData extrema_data = 3;
```

Comment:
TODO: ExtremaData isn't populated or consumed in the SDKs (afaik), and Gauges aren't represented in these protos.